packages/diagram/lib/diagram.dart

```diff
@@ -0,0 +1,442 @@
+  })  : assert(child != null),
+        assert(boundaryKey != null),
+        super(key: key);
+
+  final GlobalKey boundaryKey;
+  final Widget child;
+
+  _DiagramState createState() => new _DiagramState();
+}
+
+class _DiagramState extends State<_Diagram> {
+  Widget build(BuildContext context) {
+    return new MaterialApp(
+      debugShowCheckedModeBanner: false,
+      home: new Material(
+        child: new Builder(
+          builder: (BuildContext context) {
+            return new Center(
+              child: new RepaintBoundary(
+                key: widget.boundaryKey,
+                child: widget.child,
+              ),
+            );
+          },
+        ),
+      ),
+    );
+  }
+}
+
+const Size _kDefaultDiagramViewportSize = const Size(1280.0, 1024.0);
+
+// View configuration that allows diagrams to not match the physical dimensions
+// of the device. This will change the view used to display the flutter surface
+// so that the diagram surface fits on the device, but it doesn't affect the
+// captured image pixels.
+class _DiagramViewConfiguration extends ViewConfiguration {
+  _DiagramViewConfiguration({
+    double pixelRatio: 1.0,
+    Size size: _kDefaultDiagramViewportSize,
+  })  : _paintMatrix = _getMatrix(size, ui.window.devicePixelRatio),
+        _hitTestMatrix = _getMatrix(size, 1.0),
+        super(size: size);
+
+  static Matrix4 _getMatrix(Size size, double devicePixelRatio) {
+    final double inverseRatio = devicePixelRatio / ui.window.devicePixelRatio;
+    final double actualWidth = ui.window.physicalSize.width * inverseRatio;
+    final double actualHeight = ui.window.physicalSize.height * inverseRatio;
+    final double desiredWidth = size.width;
+    final double desiredHeight = size.height;
+    double scale, shiftX, shiftY;
+    if ((actualWidth / actualHeight) > (desiredWidth / desiredHeight)) {
+      scale = actualHeight / desiredHeight;
+      shiftX = (actualWidth - desiredWidth * scale) / 2.0;
+      shiftY = 0.0;
+    } else {
+      scale = actualWidth / desiredWidth;
+      shiftX = 0.0;
+      shiftY = (actualHeight - desiredHeight * scale) / 2.0;
+    }
+    final Matrix4 matrix = new Matrix4.compose(
+        new Vector3(shiftX, shiftY, 0.0), // translation
+        new Quaternion.identity(), // rotation
+        new Vector3(scale, scale, 1.0) // scale
+        );
+    return matrix;
+  }
+
+  final Matrix4 _paintMatrix;
+  final Matrix4 _hitTestMatrix;
+
+  @override
+  Matrix4 toMatrix() => _paintMatrix.clone();
+
+  /// Provides the transformation matrix that converts coordinates in the test
+  /// coordinate space to coordinates in logical pixels on the real display.
+  ///
+  /// This is essentially the same as [toMatrix] but ignoring the device pixel
+  /// ratio.
+  ///
+  /// This is useful because pointers are described in logical pixels, as
+  /// opposed to graphics which are expressed in physical pixels.
+  Matrix4 toHitTestMatrix() => _hitTestMatrix.clone();
```

Comment:
i'm guessing you don't actually need toHitTestMatrix ?

Reply:
Well, probably not, but I figured I'd include it in case it's needed to simulate a tap (I'm thinking of your diagram that shows the ripple animation).

Reply:
nothing seems to use it.

Reply:
Good point. Removed.